sections/earn/StakeGrid.tsx

```diff
@@ -18,14 +27,66 @@ const TimeRemainingData = () => {
 };
 
 const StakeGrid = () => {
+	const { t } = useTranslation();
 	const dispatch = useAppDispatch();
-	const earnedRewards = useAppSelector(({ earn }) => earn.earnedRewards);
+	const earnedRewards = useAppSelector(selectEarnedRewards);
+	const kwentaAmount = useAppSelector(selectKwentaAmount);
+	const wethAmount = useAppSelector(selectWethAmount);
+	const lpTotalSupply = useAppSelector(selectLPTotalSupply);
 	const yieldPerDay = useAppSelector(selectYieldPerDay);
+	const [tokenPrices, setTokenPrices] = useState<any>(null);
 
 	const handleClaim = useCallback(() => {
 		dispatch(claimRewards());
 	}, [dispatch]);
 
+	useEffect(() => {
+		const tokens = async () => {
+			const coinGeckoPrices = await sdk.exchange.batchGetCoingeckoPrices(
+				[KWENTA_COINGECKO_ADDRESS, ETH_COINGECKO_ADDRESS],
+				true
+			);
+
+			const kwentaPrice = coinGeckoPrices
+				? toWei(coinGeckoPrices[KWENTA_COINGECKO_ADDRESS]?.usd?.toString())
+				: zeroBN;
+
+			const wethPrice = coinGeckoPrices
+				? toWei(coinGeckoPrices[ETH_COINGECKO_ADDRESS]?.usd?.toString())
+				: zeroBN;
+
+			setTokenPrices({
+				kwentaPrice: kwentaPrice,
+				wethPrice: wethPrice,
+			});
+		};
+
+		(async () => {
+			await tokens();
```

Comment:
I think it would be good to move this to redux state and actions. We could add a 
```
coinGecko: {
  weth: string,
  kwenta: string
}
```
to `PricesState` which would hold these. And then add an action `fetchEarnTokenPrices`  which will handle making the request, along with reducer for setting them. 

We'll probably want to add a `usePollAction('fetchStakingTokenPrices', fetchStakingTokenPrices, 30000)` in this file so we keep fetching the prices regularly

Reply:
Thanks. Will make the changes.

Reply:
Thanks for the guidance. I moved all the calculation logic of APR (including LP token price, LP token amount, etc.) into `state/earn`.